nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/ftp/TestVirtualFileSystemView.java

```diff
@@ -0,0 +1,223 @@
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory("/");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testChangeToUnspecifiedDirectory() throws FtpException {
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory("");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testChangeToSameDirectory() throws FtpException {
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory(".");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testChangeToSameDirectoryNonRoot() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1");
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory(".");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/Directory1");
+    }
+
+    @Test
+    public void testChangeToParentDirectory() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1");
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory("..");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testChangeToParentDirectoryNonRoot() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1");
+        fileSystemView.changeWorkingDirectory("SubDirectory1");
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory("..");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/Directory1");
+    }
+
+    @Test
+    public void testChangeToNonExistentDirectory() throws FtpException {
+        // GIVEN
+
+        // WHEN
+        boolean changeDirectoryResult = fileSystemView.changeWorkingDirectory("/Directory2/SubDirectory3/SubSubDirectory");
+
+        // THEN
+        assertFalse(changeDirectoryResult);
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testGetFileAbsolute() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1/SubDirectory1");
+
+        // WHEN
+        FtpFile file = fileSystemView.getFile("/Directory2/SubDirectory3");
+
+        // THEN
+        assertEquals("/Directory2/SubDirectory3", file.getAbsolutePath());
+    }
+
+    @Test
+    public void testGetFileNonAbsolute() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1/SubDirectory1");
+
+        // WHEN
+        FtpFile file = fileSystemView.getFile("SubSubDirectory");
+
+        // THEN
+        assertEquals("/Directory1/SubDirectory1/SubSubDirectory", file.getAbsolutePath());
+    }
+
+    private User createUser() {
```

Comment:
As the user object is not really utilised in many places, a mock would be simpler